src/Task/ValidationErrorStrings.cpp

```diff
@@ -47,7 +47,7 @@ static_assert(ARRAY_SIZE(validation_error_strings) == unsigned(TaskValidationErr
               "Wrong array size");
 
 const TCHAR*
-getTaskValidationErrors(const TaskValidationErrorSet v)
+getTaskValidationErrors(const TaskValidationErrorSet &v)
```

Comment:
..... why oh why?
I don't understand the commit message.

Reply:
From my understanding, passing a reference to a subfunction is better than copying the entire object. But of course that's only true if the caller can't destroy the object while it's still being used in the subfunction (as can happen in multithreaded applications though). Is that the case here? Then, of course, I immediately withdraw my amendment!

Reply:
> From my understanding, passing a reference to a subfunction is better than copying the entire object.

But ... why is that your understanding? When I ask "why", an answer like "because that's my understanding" is never a satisfying answer. It doesn't explain anything at all.

I feared that you would reply with something like that, but when I ask "why" questions, I always want to first assume that you have good reasons, and that you can explain them. But in this case, again, I don't think you have good reasons, only some misguided (corporate?) C++ policy somebody taught you, but you never questioned it. Free yourself of this misguided policies, boot your brain and start thinking clearly.

Take a step back, and look what `TaskValidationErrorSet` really is. Did you look it up when you wrote that code? (`git grep` is your friend, it presents you the definition in a millisecond or so):
https://github.com/XCSoar/XCSoar/blob/299a01a61cb46c5743599d4ac0650ae1478ac9d6/src/Engine/Task/Factory/ValidationError.hpp#L57
And... what's this `EnumBitSet`?
https://github.com/XCSoar/XCSoar/blob/299a01a61cb46c5743599d4ac0650ae1478ac9d6/src/util/EnumBitSet.hxx#L37-L47

That's it - it's only a trivial wrapper for a 32 bit integer. In your imagination, passing a 64 bit pointer to a 32 bit integer is better than passing the 32 bit integer.
You didn't notice how poor of an idea that is because, I believe, you didn't know what you were doing, you didn't know it's just a 32 bit integer.

In the caller code, that 32 bit integer is probably already in a CPU register. After your patch, the compiler has to change all callers to allocate 4 bytes of stack space and copy the CPU register to the stack allocation, and assign the address of that stack allocation to that CPU register, to implement the new calling convention of the function you changed. Can you understand how messed up this overhead is you are introducing, for no reason?

Some objects should not be passed by value (if they are not consumed by the callee (since C++11)), but there's no unconditional rule or policy to do that for all classes. Use the grey matter in your head to decide that. References to trivial objects that can be stored in just one CPU register almost never make sense as a reference/pointer. This here is one, and it doesn't make sense.

Until 9 years ago, `TaskValidationErrorSet` was  a `std::vector`, and that object should be passed by reference. And indeed it was. It was my commit which changed this to a trivial integer which also removed the reference from the `getTaskValidationErrors()` parametr. Look: https://github.com/XCSoar/XCSoar/commit/21cc9ab4d80b4aaf788278c1ffa9e512b3b10665#diff-e5fdef1541b218e40e141c2d2bf831605f83ebb0be9c5a24da27075120f7b18a

Removing the reference was my explicit decision 9 years ago, and your commit now attempts to revert that decision.